webapp/app.py

```diff
@@ -2,25 +2,60 @@
 A Flask application for maas.io
 """
 
+import os
 import flask
 
+from canonicalwebteam.discourse_docs import DiscourseDocs, DiscourseAPI
+from canonicalwebteam.discourse_docs.parsers import DocParser
 from canonicalwebteam.flask_base.app import FlaskBase
 from canonicalwebteam.templatefinder import TemplateFinder
+from canonicalwebteam.search import build_search_view
 
 from webapp.feeds import get_rss_feed
 
 
+DISCOURSE_BASE_URL = "https://discourse.maas.io/"
+DOCS_TOPIC_ID = 25
+DOCS_CATEGORY_ID = 5
+DOCS_URL_PREFIX = "/docs"
+DOCS_TEMPLATE_PATH = "docs/document.html"
+
 app = FlaskBase(
     __name__,
     "maas.io",
     template_folder="../templates",
     static_folder="../static",
 )
 
+app.config["SEARCH_API_KEY"] = os.getenv("SEARCH_API_KEY")
+app.config["SEARCH_API_URL"] = "https://www.googleapis.com/customsearch/v1"
+app.config["SEARCH_CUSTOM_ID"] = "009048213575199080868:i3zoqdwqk8o"
+
 template_finder_view = TemplateFinder.as_view("template_finder")
 app.add_url_rule("/", view_func=template_finder_view)
 app.add_url_rule("/<path:subpath>", view_func=template_finder_view)
 
+discourse_api = DiscourseAPI(base_url=DISCOURSE_BASE_URL)
+if app.debug:
+    discourse_api.session.adapters["https://"].timeout = 99
+
+discourse_docs = DiscourseDocs(
+    api=discourse_api,
+    index_topic_id=DOCS_TOPIC_ID,
+    category_id=DOCS_CATEGORY_ID,
+    document_template=DOCS_TEMPLATE_PATH,
+    url_prefix=DOCS_URL_PREFIX,
+)
+discourse_docs.init_app(app)
+
+
+# Search
+app.add_url_rule(
+    "/docs/search",
```

Comment:
just to be picky :P

maybe `DOCS_URL_PREFIX + /search` ?

Reply:
Why does `DOCS_URL_PREFIX` exist?

Reply:
Some sources for my point of view:

https://en.wikipedia.org/wiki/Convention_over_configuration
https://medium.com/@rdsubhas/10-modern-software-engineering-mistakes-bc67fbef4fc8#97ef

Reply:
Where are you seeing `DOCS_URL_PREFIX`? `rg DOCS_URL_PREFIX` returns nothing for me.

Reply:
https://github.com/canonical-web-and-design/maas.io/pull/346/files#diff-bed92ddcb62003ca6b223dfa86df23b3R20

(Line 20 of this file)

Reply:
Sorry I did find it. Yeah I can see that since it existed,I should use it. So I removed it - problem solved =)

Reply:
I don't really mind not using the prefix in this way. But then all those variables that are defined could be removed since only used in the DiscourseDocs class

Reply:
Yep, thanks for bringing them to my attention. I also removed all the redundant search config =)